### PR TITLE
cmake: add feature summary

### PR DESCRIPTION
**Pull request by junghans**
Only documented the most important features

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -77,12 +79,15 @@ if(Cabana_REQUIRE_HDF5)
 else()
   Cabana_add_dependency( PACKAGE MPI COMPONENTS CXX )
 endif()
+set_package_properties(MPI PROPERTIES TYPE RECOMMENDED PURPOSE "Used for parallization")
```
Comment (dalg24) [inline CMakeLists.txt:82]:
```suggestion
set_package_properties(MPI PROPERTIES TYPE RECOMMENDED PURPOSE "Used for parallelization")
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -118,6 +125,7 @@ if(CMAKE_VERSION VERSION_LESS 3.26)
 else()
   Cabana_add_dependency( PACKAGE HDF5 COMPONENTS C )
 endif()
+set_package_properties(HDF5 PROPERTIES TYPE RECOMMENDED PURPOSE "Used for I/O")
```
[the remark this answers is not in the document]
Comment (junghans) [reply CMakeLists.txt:128]: Whether we feel strong about a dependency or not. It is our call.

[the remark this answers is not in the document]
Comment (junghans) [reply CMakeLists.txt:128]: From the cmake documentation: _"In this case it is a package which can be used by the project when available at buildtime, but it also work without. RECOMMENDED is similar to OPTIONAL, i.e. the project will build if the package is not present, but the functionality of the resulting binaries will be severely limited."_

File: CMakeLists.txt [hunk 3]
```diff
@@ -92,9 +97,11 @@ Cabana_add_dependency( PACKAGE HYPRE VERSION 2.22.1 )
```
Comment (streeve) [inline CMakeLists.txt:97]: set_package_properties(HYPRE PROPERTIES TYPE OPTIONAL PURPOSE "Used for structured solves")
```suggestion

```

File: CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (streeve) [reply CMakeLists.txt:128]: I like MPI as recommended, but I would go with optional for hdf5

**Comment by junghans** [comment]
@streeve can you add something for `HYPRE`?

**Review by streeve — APPROVED** [review]

**Comment by streeve** [comment]
> @streeve can you add something for `HYPRE`?

Done, I messed up the suggestion..

**Comment by junghans** [comment]
I only documented the `Cabana_ENABLE_GRID` option, should we do other?
I think coverage, asm build etc and are not really user facing, so we can ignore that.

**Comment by streeve** [comment]
> I only documented the `Cabana_ENABLE_GRID` option, should we do other? I think coverage, asm build etc and are not really user facing, so we can ignore that.

Agreed, I think all the require options would be a bit verbose. Added test/example/benchmarks

**Comment by junghans** [comment]
@dalg24-jenkins re-test this!

**Comment by streeve** [comment]
> @dalg24-jenkins re-test this!

We still need to get this working again....

**Pull request merged**